### ISSUE TITLE
TestActivityExtension lifecycle problems with more than a single test case

### ISSUE THREAD

**Issue by mreardonx** [temporalio/sdk-java#928]
## Expected Behavior
The Junit 5 TestActivityExtension lifecycle should support multiple test methods in a single test class.

## Actual Behavior
With two test methods, the first test method completes as expected, but the second test method throws an exception when an activity task is submitted.

```
Task java.util.concurrent.FutureTask@39b85002 rejected from java.util.concurrent.ThreadPoolExecutor@377fb945[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 1]
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@39b85002 rejected from java.util.concurrent.ThreadPoolExecutor@377fb945[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 1]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
	at java.util.concurrent.Executors$DelegatedExecutorService.submit(Executors.java:681)
	at io.temporal.testing.TestActivityEnvironmentInternal$TestActivityExecutor.executeActivity(TestActivityEnvironmentInternal.java:345)
	at io.temporal.testing.TestActivityEnvironmentInternal$TestActivityExecutor.executeActivity(TestActivityEnvironmentInternal.java:298)
	at io.temporal.internal.sync.ActivityStubImpl.executeAsync(ActivityStubImpl.java:49)
	at io.temporal.internal.sync.ActivityStubBase.execute(ActivityStubBase.java:38)
	at io.temporal.internal.sync.ActivityInvocationHandler.lambda$getActivityFunc$0(ActivityInvocationHandler.java:77)
	at io.temporal.internal.sync.ActivityInvocationHandlerBase.invoke(ActivityInvocationHandlerBase.java:70)
	at io.temporal.internal.sync.DeterministicRunnerWrapper.lambda$invoke$0(DeterministicRunnerWrapper.java:43)
	at io.temporal.internal.sync.CancellationScopeImpl.run(CancellationScopeImpl.java:101)
	at io.temporal.internal.sync.WorkflowThreadImpl$RunnableWrapper.run(WorkflowThreadImpl.java:107)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

## Steps to Reproduce the Problem

  1. Modifiy the io.temporal.samples.hello.HelloDynamicActivityJUnit5Test test to have two test methods by making of copy of `testDynamicActivity` method and renaming it.
  2. Run the unit test.  Both test cases should have passed, but only one will.

## Specifications

  - Version:  1.6.0
  - Platform: Java 17, Junit 5

**Comment by eeshvardasikcm** [comment]
this bug doesn't look too difficult to cure.

I want to get in on the activity packages.
haven't had a chance to read the contributing guide. busy making my own way.

I believe this is Apache license isn't it. I can give some recommendation on how to fix but I'm not doing pull requests. I'm only doing comments.

**Comment by eeshvardasikcm** [comment]
so I suppose the work around of using only one test case per class is not sufficient here.

**Comment by eeshvardasikcm** [comment]
> another look around maybe is using a reduced version of J-Unit or a reduced version of Java API.
> 
> have these options been explored or are these actually the goals of temporal to reach compatibility with J-Unit 5 and Java 17?

contribution guidelines show Java 8. Java 17 is relatively new.

have these tests been attempted using Java 8?

maybe this issue will make more sense as a new feature?

**Comment by eeshvardasikcm** [comment]
a clear solution has been outlined in the latest release

"Migration notes
temporal-testing-junit4, temporal-testing-junit5, and temporal-testing merged into one temporal-testing module that exposes Gradle "capabilities."
For Gradle:
See https://github.com/temporalio/sdk-java/tree/master/temporal-testing#usage for Gradle-specific instructions. Or you can just switch to temporal-testing and provide junit dependency yourself without digging into Gradle's capabilities.
For other build tools:
Just switch onto using just temporal-testing instead of temporal-testing-junit4 or temporal-testing-junit5."

- does the solution to switch on to just temporal testing instead of temporal testing J-Unit 5 work?

- or is there a problem with the new migration?

**Comment by Bennett-Lynch** [comment]
+1

**Comment by eeshvardasikcm** [comment]
I didn't get a response and noticed my comments were deleted. That's not very honorable of the gift I gave. Did I violate some commenting rules? I may not have read all of your rules.

So it appears you are going with a single solution instead of the architectural improvement that I indicated. It's not my money your spending so I'm not losing money on this fix.

**Comment by Spikhalskiy** [comment]
@eeshvardasikcm 

> I didn't get a response and noticed my comments were deleted. That's not very honorable of the gift I gave. Did I violate some commenting rules? I may not have read all of your rules.
We don't guarantee responses and preservation of all the comments. 
I found the comments irrelevant to the discussion and the issue and there were 5 of them, one less applicable than another.
Taking into account that there was a valuable up-to-the-point contribution from another community member later, I decided to trim several wordy irrelevant comments to keep the history of this issue and what actually happened with it making sense for the next reader.
We appreciate all contributions and efforts, but we leave the right to accept or reject them.

> That's not very honorable of the gift I gave.
I'm sorry, we had to reject it.

> So it appears you are going with a single solution instead of the architectural improvement that I indicated. 
Correct, this issue doesn't need any architectural changes, it's a pretty trivial mistake.

> It's not my money your spending so I'm not losing money on this fix.
Sounds great.
